packages/fxa-auth-server/lib/metrics/glean/index.ts

```diff
@@ -109,13 +110,54 @@ const createEventFn =
         utm_term: metricsContext.utmTerm || '',
       };
 
+      // TODO: which events should have a reason?
+      const eventReason = metricsData?.reason || '';
```

Comment:
We used 'reason' because we want something generic since the "event" itself was generic.  Now that we want to use event metrics, we can have specific extra keys.  Although that might make backporting data a little more difficult?

I guess to really answer your question: we need to check the call sites where events are emitted to know which pass in a reason.

Reply:
I passed the reason to events that are using it.

Reply:
@akkomar would backporting data be more difficult if we change the name of the extra key to something event-specific?

Reply:
I don't think so, we can do that.